--- src.orig/sys/sys/namei.h	2017-09-28 13:57:40.727911673 +0300
+++ src.zad6/sys/sys/namei.h	2017-09-28 23:14:58.280981947 +0300
@@ -131,6 +131,7 @@
  * name being sought. The caller is responsible for releasing the
  * buffer and for vrele'ing ni_startdir.
  */
+#define ISUFSRENAME     0x00000001 /* mask for ufs rename */
 #define	RDONLY		0x00000200 /* lookup with read-only semantics */
 #define	HASBUF		0x00000400 /* has allocated pathname buffer */
 #define	SAVENAME	0x00000800 /* save pathname buffer */
