--- /root/src.orig/sys/netinet/ip_fastfwd.c	2017-11-17 21:54:43.670031078 +0300
+++ /root/src.zad6/sys/netinet/ip_fastfwd.c	2017-11-17 23:42:09.834639632 +0300
@@ -171,13 +171,18 @@
 	int error = 0;
 	int hlen, mtu;
 	struct m_tag *fwd_tag = NULL;
-
+	
+//	printf("\nip_fastforwar");
 	/*
 	 * Are we active and forwarding packets?
 	 */
 	if (!V_ipfastforward_active || !V_ipforwarding)
 		return m;
 
+	if (m->m_pkthdr.len > 1100){
+		goto drop;
+	}
+
 	M_ASSERTVALID(m);
 	M_ASSERTPKTHDR(m);
 
@@ -205,7 +210,10 @@
 	}
 
 	ip = mtod(m, struct ip *);
-
+	
+//	printf("\nip->ip_len: {%d}", ip->ip_len);
+//	printf("\nip->ip_hl: {%d}", ip->ip_hl);
+//	printf("\nm->m_pkthdr.len: {%d}", m->m_pkthdr.len);
 	/*
 	 * Is it IPv4?
 	 */
