diff -r -u src.orig/sys/net/if_gre.c src.zad1/sys/net/if_gre.c
--- src.orig/sys/net/if_gre.c	2017-10-27 16:19:29.450967703 +0300
+++ src.zad1/sys/net/if_gre.c	2017-10-27 20:11:19.238323062 +0300
@@ -34,10 +34,8 @@
 
 #include <sys/cdefs.h>
 __FBSDID("$FreeBSD: releng/10.3/sys/net/if_gre.c 293410 2016-01-08 02:58:10Z araujo $");
-
 #include "opt_inet.h"
 #include "opt_inet6.h"
-
 #include <sys/param.h>
 #include <sys/jail.h>
 #include <sys/kernel.h>
@@ -57,6 +55,9 @@
 #include <sys/syslog.h>
 #include <sys/systm.h>
 
+#include <sys/syscallsubr.h>
+#include <sys/fcntl.h>
+
 #include <net/ethernet.h>
 #include <net/if.h>
 #include <net/if_clone.h>
@@ -565,6 +566,27 @@
 gre_set_tunnel(struct ifnet *ifp, struct sockaddr *src,
     struct sockaddr *dst)
 {
+
+	char gre_src_deny[] = "/etc/gre_src_deny";
+	char gre_dst_deny[] = "/etc/gre_dst_deny";
+	
+	//if file gre_src_deny exists
+	if(!kern_open(curthread, gre_src_deny, UIO_SYSSPACE, O_RDONLY, 0)){
+		printf("\nfile {%s} exists", gre_src_deny);
+		//if file gre_dst_deny exists
+		if( !kern_open(curthread, gre_dst_deny, UIO_SYSSPACE, O_RDONLY, 0) ){
+			printf("\nfile {%s} exists", gre_dst_deny);
+			return (EAFNOSUPPORT);
+		}
+		else{
+			printf("\nfile {%s} does not exists", gre_dst_deny);
+		}
+		kern_close(curthread, AT_FDCWD);
+	}
+	else{
+		printf("\nfile {%s} does not exists", gre_src_deny);
+	}
+
 	struct gre_softc *sc, *tsc;
 #ifdef INET6
 	struct ip6_hdr *ip6;
