--- src.orig/sys/ufs/ufs/ufs_vnops.c	2017-09-28 13:57:48.075958861 +0300
+++ src.zad3/sys/ufs/ufs/ufs_vnops.c	2017-09-28 20:38:26.524510945 +0300
@@ -281,6 +281,15 @@
 	if ((ip->i_flags & APPEND) &&
 	    (ap->a_mode & (FWRITE | O_APPEND)) == FWRITE)
 		return (EPERM);
+
+	//flag sticky bit and this not directory
+        if (vp->v_type != VDIR && 
+		(ip->i_mode & ISVTX) &&
+                (ap->a_mode & FWRITE) &&
+                (ip->i_uid == ap->a_cred->cr_uid)){
+                return (EPERM);
+        }
+
 	vnode_create_vobject(vp, DIP(ip, i_size), ap->a_td);
 	return (0);
 }
@@ -488,8 +497,8 @@
 		vap->va_size = ip->i_din2->di_size;
 		vap->va_mtime.tv_sec = ip->i_din2->di_mtime;
 		vap->va_mtime.tv_nsec = ip->i_din2->di_mtimensec;
-		vap->va_ctime.tv_sec = ip->i_din2->di_ctime;
-		vap->va_ctime.tv_nsec = ip->i_din2->di_ctimensec;
+		vap->va_ctime.tv_sec = ip->i_din2->di_birthtime;
+		vap->va_ctime.tv_nsec = ip->i_din2->di_birthnsec;
 		vap->va_birthtime.tv_sec = ip->i_din2->di_birthtime;
 		vap->va_birthtime.tv_nsec = ip->i_din2->di_birthnsec;
 		vap->va_bytes = dbtob((u_quad_t)ip->i_din2->di_blocks);
@@ -519,7 +528,18 @@
 	struct ucred *cred = ap->a_cred;
 	struct thread *td = curthread;
 	int error;
-
+//check, that this ACL
+#ifdef UFS_ACL
+	//check nfsv4ACL
+        if ((vp->v_mount->mnt_flag & MNT_NFS4ACLS)
+		//check IMMUTABLE flag
+                && ((vap->va_flags & (SF_IMMUTABLE | UF_IMMUTABLE)) != 0)) {
+		//print in dmesg
+		printf("\nError operation denied on check IMMUTABLE FLAG");
+		//return error "Operation not permitted"
+                return (EPERM);
+        }
+#endif
 	/*
 	 * Check for unsettable attributes.
 	 */
